packages/providers/src/core/PoapCollectionsApi/PoapCollectionsApi.ts

```diff
@@ -0,0 +1,112 @@
+import { PostCollectionsInput } from './../../ports/CollectionsApiProvider/types/PostCollectionsInput';
+import { CollectionResponse } from './../../ports/CollectionsApiProvider/types/CollectionResponse';
+import { PutCollectionsInput } from '../../ports/CollectionsApiProvider/types/PutCollectionsInput';
+import { CollectionsApiProvider } from './../../ports/CollectionsApiProvider/CollectionsApiProvider';
+
+const DEFAULT_COLLECTIONS_BASE_URL = 'https://collections.poap.tech';
+/**
+ * The `PoapCollectionsApi` class provides methods to interact with the POAP Collections API,
+ * allowing for the creation and modification of POAP collections.
+ *
+ * It requires an authToken for authorization and uses a provided or default base URL for API requests.
+ *
+ * @class PoapCollectionsApi
+ * @implements {CollectionsApiProvider}
+ */
+export class PoapCollectionsApi implements CollectionsApiProvider {
+  private baseUrl: string;
+
+  /**
+   * Creates an instance of PoapCollectionsApi.
+   *
+   * @constructor
+   * @param {string} [baseUrl='https://collections.poap.tech'] - The base URL for the Collections API.
+   * If not provided, a default URL is used.
+   */
+  constructor(baseUrl: string) {
```

Comment:
Better to keep using the props with baseUrl as a property as in other places so we keep consistency.